### PR TITLE
Extract interface for management.Client

### PR DESCRIPTION
**Pull request by paulmey**
Allows tests to stub out at the HTTP client level.

cc: @aznashwan @ahmetalpbalkan

### CODE REVIEW

File: management/client.go [hunk 1]
```diff
@@ -25,6 +25,13 @@ type Client struct {
 	config          ClientConfig
 }
 
+type AzureHttpClient interface {
+	SendAzureGetRequest(url string) ([]byte, error)
```
Comment (ahmetalpbalkan) [inline management/client.go]: Call this Client. And rename Client to client.

Comment (paulmey) [inline management/client.go]: yep, good suggestion

**Comment by paulmey** [comment]
how about this @ahmetalpbalkan

**Comment by paulmey** [comment]
It's actually quite convenient for making decorators this way: https://github.com/paulmey/azure-sdk-for-go/commit/0e7de0d2cc56ac52396bd94e283b9b481f1fda62

**Comment by ahmetalpbalkan** [comment]
Would like to see @aznashwan's input before merging.

**Comment by aznashwan** [comment]
@ahmetalpbalkan looks great.

This makes the client itself patchable, which is great.
I am, however, a bit concerned about the logistical aspects of up-and-coming testing; for which I will follow-up with an email when I get to that hurdle.

Thanks very much for the work guys! :smiley:

**Comment by ahmetalpbalkan** [comment]
Thanks for your input! Merging.

**Pull request merged**